### PR TITLE
[MAINTENANCE] Fix Type-Checking steps

### PR DESCRIPTION
**Pull request by Kilo59**
## Changes proposed in this pull request:
- Fixes type-checking step so that it blocks PR's
- Split type-checking into two steps
   - normal type-checking
   - type-checking of stub source files

### CODE REVIEW

File: great_expectations/compatibility/sqlalchemy_and_pandas.py [hunk 1]
```diff
@@ -38,18 +37,14 @@ def execute_pandas_reader_fn(
...
             warn_pandas_less_than_2_0_and_sqlalchemy_greater_than_or_equal_2_0()
         with warnings.catch_warnings():
             warnings.filterwarnings(action="ignore", category=RemovedIn20Warning)
             reader_fn_result: pd.DataFrame | list[pd.DataFrame] = reader_fn(
                 **reader_options
             )
     else:
-        reader_fn_result: pd.DataFrame | list[pd.DataFrame] = reader_fn(
-            **reader_options
-        )
```
Comment (Kilo59) [inline great_expectations/compatibility/sqlalchemy_and_pandas.py]: @anthonyburdi please check my logic here but this should fix the error we get in the quickstart when SQLAlchemy isn't installed

File: ci/azure-pipelines-dev.yml [hunk 2]
```diff
@@ -152,9 +152,11 @@ stages:
       - script: |
           pip install --requirement requirements-types.txt
           invoke type-check --ci --pretty
+        name: StaticTypeCheck
+      - script: |
           # initial run doesn't check `.pyi` source files
           invoke type-check --ci --pretty --check-stub-sources
-        name: StaticTypeCheck
+        name: TypeCheckStubSourceFiles
```
Comment (Kilo59) [inline ci/azure-pipelines-dev.yml:159]:
```suggestion
        name: CheckStubSourceFiles
```

File: great_expectations/compatibility/sqlalchemy_and_pandas.py (hunk 1 above)
Comment (Kilo59) [reply great_expectations/compatibility/sqlalchemy_and_pandas.py]: I've reverted this stuff

**Review by anthonyburdi — APPROVED** [review]
Thank you!

**Pull request merged**